docs/input.md

```diff
@@ -93,8 +93,10 @@ On Linux you can use `lsusb` for step 2. Step 3 involves adding a udev rule on
 most linux distributions. The udev rule (/etc/udev/rules.d/999-xbox-
 gamepad.rules) for a Controller-S could look like this:
 
-	SUBSYSTEMS=="usb", ATTRS{idVendor}=="045e", ATTRS{idProduct}=="0288", GROUP="users", MODE="660" # Hub
-	SUBSYSTEMS=="usb", ATTRS{idVendor}=="045e", ATTRS{idProduct}=="0289", GROUP="users", MODE="660" # Gamepad
+	# Hub
+	SUBSYSTEMS=="usb", ATTRS{idVendor}=="045e", ATTRS{idProduct}=="0288", GROUP="users", MODE="660"
+	# Gamepad
+	SUBSYSTEMS=="usb", ATTRS{idVendor}=="045e", ATTRS{idProduct}=="0289", GROUP="users", MODE="660"
```

Comment:
For a more complete variant, this is my local file:

```
# Duke (Hub; Gamepad)
SUBSYSTEMS=="usb", ATTRS{idVendor}=="045e", ATTRS{idProduct}=="001c", GROUP="users", MODE="660"
SUBSYSTEMS=="usb", ATTRS{idVendor}=="045e", ATTRS{idProduct}=="0202", GROUP="users", MODE="660"

# Controller S (Hub; Gamepad)
SUBSYSTEMS=="usb", ATTRS{idVendor}=="045e", ATTRS{idProduct}=="0288", GROUP="users", MODE="660"
SUBSYSTEMS=="usb", ATTRS{idVendor}=="045e", ATTRS{idProduct}=="0289", GROUP="users", MODE="660"
```

*(Note: I haven't tested this udev rule in months)*

This also matches with public lists like https://devicehunt.com/view/type/usb/vendor/045E/device/0288.

There are still many versions of those gamepads missing of-course.

Reply:
I'd say it's safe to assume most of the user base will be covered by these two. :)
PR updated with your list.